craft_providers/actions/snap_installer.py

```diff
@@ -111,6 +153,15 @@ def inject_from_host(*, executor: Executor, snap_name: str, classic: bool) -> No
     :raises SnapInstallationError: on unexpected error.
     """
     target_snap_path = pathlib.Path(f"/tmp/{snap_name}.snap")
+    host_revision = _get_host_snap_revision(snap_name=snap_name)
+    target_revision = _get_target_snap_revision(snap_name=snap_name, executor=executor)
+
+    if target_revision is not None and target_revision == host_revision:
+        logger.debug(
```

Comment:
It may be useful to log which version that is.

Furthermore, in these cases I've found that when something weird happens around this mechanism, the first thing you need to understand the situation are the versions found in both places (host and target). So I would directly always log both values (in debug, of course, with repr) and then another line if skipped.